dask/dataframe/core.py

```diff
@@ -2630,12 +2630,21 @@ def _is_single_last_div(x):
 
 def repartition_npartitions(df, npartitions):
     """ Repartition dataframe to a smaller number of partitions """
-    npartitions = min(npartitions, df.npartitions)
-    k = int(math.ceil(df.npartitions / npartitions))
-    divisions = df.divisions[::k]
-    if len(divisions) <= npartitions:
-        divisions = divisions + (df.divisions[-1],)
-    return df.repartition(divisions=divisions)
+    size = df.npartitions / npartitions
+    L = [int(i * size) for i in range(npartitions + 1)]
+    name = 'repartition-%d-%s' % (npartitions, df._name)
```

Comment:
Should I use `tokenize(df)` or `tokenize(df._name)` or something else? I copied this from `Bag.repartition`: https://github.com/dask/dask/blob/master/dask/bag/core.py#L1060-L1079. Thus, changing this needs to be done there also, isn't it?